Runtime/AvatarCreator/UI/Elements/TemplateAvatarElement.cs

```diff
@@ -0,0 +1,94 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class TemplateAvatarElement : MonoBehaviour
+    {
+        private const string TAG = nameof(TemplateAvatarElement);
+
+        [Header("UI Elements")]
+        [Space(5)]
+        [SerializeField] private ButtonElement buttonElementPrefab;
+        [SerializeField] private Transform buttonContainer;
+        [SerializeField] private GameObject selectedIcon;
+
+        [Header("Events")]
+        [Space(5)]
+        public UnityEvent<TemplateData> onTemplateSelected;
+
+        private List<TemplateData> avatarTemplateDataList;
+        private ButtonElement[] templateAvatarButtons;
+        private TemplateFetcher templateFetcher;
+
+        private void Awake()
+        {
+            templateFetcher = new TemplateFetcher();
+        }
+
+        public async void LoadAndCreateButtons()
+        {
+            await LoadTemplateRenders();
+            CreateButtons();
+        }
+
+        public async void FetchTemplateData()
+        {
+            avatarTemplateDataList = await templateFetcher.GetTemplates();
+            if (avatarTemplateDataList == null || avatarTemplateDataList.Count == 0)
+            {
+                SDKLogger.LogWarning(TAG, "No templates found");
+            }
+        }
+
+        public async Task LoadTemplateRenders()
+        {
+            avatarTemplateDataList = await templateFetcher.GetTemplatesWithRenders();
+        }
```

Comment:
A bit of over complication, this way I am not sure which function to call and in which order.

Reply:
What would you suggest here instead? 
Perhaps I am going overboard with providing multiple [ublic functions, the main thing I was trying to enable was that devs could control how they want to do the loading without needing to create their own script. But maybe that isn't actually necessary. Because on the other hand, we kinda want devs to be creating their own scripts as it should (in theory) reduce the risk of issues related to package updates. 
It's kind of like utility vs simplicity 😅

Reply:
I guess you shouldn't know which function to call and in what order, it's up to you based on how you're building your experience as a partner. The purpose of our SDK is to offer utility, not to dictate the steps in which you call our API to build your game / experience.

`Because on the other hand, we kinda want devs to be creating their own scripts as it should (in theory) reduce the risk of issues related to package updates.`

We definitely want devs creating their own scripts, but this lends to having how it is currently, because those scripts would call these underlying SDK scripts. I guess the point isn't about whether they interact with the SDK through their own scripts or through the editor, but more about us offering utility that they can use in whichever means they choose. The minute a dev has to create a script to rework something in one of our lower level scripts then we've done something wrong.